server/chat-plugins/chatlog.ts

```diff
@@ -380,4 +601,28 @@ export const commands: ChatCommands = {
 		const roomid = targetRoom ? targetRoom.roomid : target;
 		this.parse(`/join view-chatlog-${roomid}--today`);
 	},
+
+	sl: 'searchlogs',
+	searchlog: 'searchlogs',
+	searchlogs(target, room) {
+		target = target.trim();
+		const [search, tarRoom, cap, date] = target.split(',') as [string, string, number, number];
+		if (!target) return this.parse('/help searchlogs');
+		if (!search) return this.errorReply('Specify a query to search the logs for.');
+		if (cap && isNaN(cap) && toID(cap) !== 'all') return this.errorReply(`Cap must be a number or [all].`);
+		const input = search.includes('|') ? search.split('|').map(item => item.trim()).join('-') : search;
+		const currentMonth = Chat.toTimestamp(new Date()).split(' ')[0].slice(0, -3);
+		const curRoom = tarRoom ? Rooms.search(tarRoom) : room;
+		const limit = cap ? `&${cap}` : `&500`;
```

Comment:
```suggestion
		const limit = cap ? `&${cap}` : `&500`;
```

`&` isn't a valid character in roomids, either :(